api/v1alpha1/mutation_types.go

```diff
@@ -11,24 +11,30 @@ import (
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// MutationObject defines any object which produces a snapshot
+// +k8s:deepcopy-gen=false
+type MutationObject interface {
```

Comment:
This only appears to be implemented by Localization right, yes? Doesn't configuration and Resource also produce snapshots?

Reply:
Oh I see this is incomplete right now, with some build failures and stuff. Okay. Okay, I'll review the structure instead.